arangod/VocBase/LogicalCollection.cpp

```diff
@@ -1263,7 +1263,7 @@ auto LogicalCollection::getDocumentState()
   using namespace replication2::replicated_state;
   auto stateMachine =
       std::dynamic_pointer_cast<ReplicatedState<document::DocumentState>>(
-          vocbase().getReplicatedStateById(shardIdToStateId(name())));
+          vocbase().getReplicatedStateById(shardIdToStateId(name())).get());
```

Comment:
Note that `.get()` throws an exception if the result is an error.

More importantly, are we sure this can't happen? For example, can the shard (including the state) be deleted before this function is executed?

In which case we must not do any production assertion here, but handle it somewhat more gracefully.

Reply:
In `ReplicatedRocksDBTransactionCollection::ensureCollection` we end up asserting for the availability of the replicated state anyway. I think we're not supposed to reach that assert, but the current implementation is temporary anyway, and we'll have to change it once we introduce `distributeShardsLike`. I have nothing against using `.get()` here, as long as we'll be able where to look in case something blows up in a CI test.

Reply:
Alright, thanks. I'll add an assertion for now, with a comment that we should check whether to remove it.